Content.Shared/_CM14/Comtech/Barbed/BarbedSystem.cs

```diff
@@ -0,0 +1,99 @@
+    }
+    public void OnInteractUsing(EntityUid uid, BarbedComponent component, InteractUsingEvent args)
+    {
+        if (component.IsBarbed == false && HasComp<BarbedwireComponent>(args.Used))
+        {
+            var doAfterEventArgs = new DoAfterArgs(EntityManager, args.User, component.WireTime, new BarbedDoAfterEvent(), uid, used: args.Used)
+            {
+                BreakOnUserMove = true,
+                BreakOnDamage = true,
+                NeedHand = true,
+            };
+            _popupSystem.PopupClient(Loc.GetString("barbed-wire-slot-wiring"), uid, args.User, PopupType.Small);
+            _doAfterSystem.TryStartDoAfter(doAfterEventArgs);
+            return;
+        }
+
+        if (component.IsBarbed == true && HasComp<BarbedwireComponent>(args.Used))
+        {
+            _popupSystem.PopupClient(Loc.GetString("barbed-wire-slot-insert-full"), uid, args.User, PopupType.Small);
+            return;
+        }
+
+        if (component.IsBarbed == true && TryComp<ToolComponent>(args.Used, out var tool))
+        {
+            if (_toolSystem.HasQuality(args.Used, "Cutting", tool))
+            {
+                _popupSystem.PopupClient(Loc.GetString("barbed-wire-cutting-action-begin"), uid, args.User, PopupType.Small);
+                var wirecutterDoAfterEventArgs = new DoAfterArgs(EntityManager, args.User, component.CutTime, new CutBarbedDoAfterEvent(), uid, used: args.Used)
+                {
+                    BreakOnUserMove = true,
+                    BreakOnDamage = true,
+                    NeedHand = true,
+                };
+                _doAfterSystem.TryStartDoAfter(wirecutterDoAfterEventArgs);
+                return;
+            }
+        }
+        return;
+    }
+    private void OnDoAfter(EntityUid uid, BarbedComponent component, BarbedDoAfterEvent args)
+    {
+        if (args.Used == null || args.Cancelled)
+            return;
+
+        if (TryComp<StackComponent>(args.Used.Value, out var stackComp))
+        {
+            _stacks.Use(args.Used.Value, 1, stackComp);
+        }
+
+        component.IsBarbed = true;
+        _appearance.SetData(uid, BarbedWireVisuals.Wired, true);
+        _popupSystem.PopupClient(Loc.GetString("barbed-wire-slot-insert-success"), uid, args.User, PopupType.Small);
+        return;
+    }
+    private void WireCutterOnDoAfter(EntityUid uid, BarbedComponent component, CutBarbedDoAfterEvent args)
+    {
+        //EntityManager.SpawnEntity(component.Spawn, Transform(uid).Coordinates); Not sure how to make it so when wirecut spawns a metal rod
+        component.IsBarbed = false;
+        _appearance.SetData(uid, BarbedWireVisuals.Wired, false);
+        _popupSystem.PopupClient(Loc.GetString("barbed-wire-cutting-action-finish"), uid, args.User, PopupType.Small);
+        return;
+    }
+    private void OnAttacked(EntityUid uid, BarbedComponent component, AttackedEvent args)
+    {
+        if (component.IsBarbed == true)
```

Comment:
Same here, don't need to specify == true since it's not a nullable boolean. If it was, you would have to.
```suggestion
        if (component.IsBarbed)
```